### DISCUSSION TITLE
rowClick data format

### DISCUSSION THREAD

**Discussion by pmkroeker** [grid-js/gridjs#749] [Q&A]
This is related to #502.

I'm curious as to why when a row is clicked the following data structure is returned:
>row: [{"isTrusted":true},{"_id":"60aeb549-8e65-4b4f-bebe-d2f441feca76","_cells":[{"_id":"25ffecb6-c68b-4635-9773-aad45bfdd0e6","data":"John"},{"_id":"635f743a-d845-4402-a0fd-0789090b0d11","data":"john@example.com"},{"_id":"af06b34e-0a93-4125-859a-5a73e1d9fdcd","data":"(353) 01 222 3333"}]}]

It would make a bit more sense to me, and be easier to work with rowClick event if the row dataset was returned as it was passed in like:
>['John', 'john@example.com', '(353) 01 222 3333']

or like

> {name: 'John', emaill: '...', phoneNumber: '...'}

or even the index number of the clicked row.

This would make performing actions on the data from that row much easier to handle, without needing an intermediate plugin to reconstruct the data.

I'm curious as to why the rowClick dataset is returned the way it is.

**Comment by raghu-bhandi** [comment]
I guess it is something to do with the way the html is generated using "preact". I notice that the data row is not available to any of the functions.
Here is my work-around using the "columns" attribute.
1. Add a column (or member) to data with the row-index as its value. 
2. In the "columns" attribute, use {hidden:true} as your first column
3. With this, idx can be fetched as row._cells[0].data

I am attaching an example
  <script>
    const grid = new gridjs.Grid({
      columns: [
      {id: 'idx', hidden: true},
      {id: 'name', name: 'Name', attributes: {'data-click': 'name-action'}},
        {id: 'age', name: 'Age', attributes: {'data-click': 'age-action'}},
        {id: 'country', name: 'Country', attributes: {'data-click': 'country-action'}},
      ],
      data: [
        {name:"John", age:25, country: "USA", idx: 0},
        {name: "Maria", age:30, country: "UK", idx: 1},
        {name: "Tom", age:35, country: "Canada", idx: 2},
        {name: "Sophia", age:28, country:"Australia", idx:3}
      ],
    });

    grid.on('rowClick', (...args) => {
      console.log('Row0clickd:', args);
      const row = args[1];
      const cell =row._cells[0];
      const idx = cell.data;
      console.log('idx=', +idx);
    });
    grid.on('cellClick', (...args) => {
      console.log('CellClicked:', args);
      const row = args[3];
      const cell =row._cells[0];
      const idx = cell.data;
      console.log('action=', args[0].target.getAttribute('data-click'));
      console.log('idx=', +idx);
    });

    // Render the grid
    grid.render(document.getElementById("wrapper"));
  </script>
`
`
